sound/soc/sof/pcm.c

```diff
@@ -344,6 +352,9 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 		/* invoke platform trigger to stop DMA even if pcm_ops isn't set or if it failed */
 		if (!pcm_ops || !pcm_ops->platform_stop_during_hw_free)
 			snd_sof_pcm_platform_trigger(sdev, substream, cmd);
+		if (pcm_ops && pcm_ops->platform_stop_during_hw_free &&
+		    cmd == SNDRV_PCM_TRIGGER_STOP)
+			spcm->stream_stopped[substream->stream] = true;
```

Comment:
a comment would really help here..
we set the stream_stopped flag to true when we have not stopped the stream?

Reply:
as far as ALSA is concerned the stream is stopped. The pipeline may not be stopped yet but thats different and thats what we need to handle when stream_stopped is set

Reply:
stopping pipelines here would be in theory the right thing to do but in order to keep the changes less invasive, we decided to keep the existing sequence. We can revisit this later if this needs to change